ibc-core/ics23-commitment/types/src/specs.rs

```diff
@@ -61,15 +66,29 @@ impl From<ProofSpecs> for Vec<RawProofSpec> {
 #[derive(Clone, Debug, PartialEq)]
 struct ProofSpec(RawProofSpec);
 
-impl From<RawProofSpec> for ProofSpec {
-    fn from(spec: RawProofSpec) -> Self {
-        Self(RawProofSpec {
-            leaf_spec: spec.leaf_spec.map(|lop| LeafOp::from(lop).0),
-            inner_spec: spec.inner_spec.map(|ispec| InnerSpec::from(ispec).0),
+impl TryFrom<RawProofSpec> for ProofSpec {
+    type Error = CommitmentError;
+    fn try_from(spec: RawProofSpec) -> Result<Self, CommitmentError> {
+        if spec.max_depth < spec.min_depth
+            || spec.min_depth < 0
+            || spec.max_depth < 0
```

Comment:
please add comments about the failure case.

example: why `0` is case is allowed.

Reply:
hey sorry for not explaining myself :sweat_smile: I know the logic. I suggested adding them in the comments.

You can refer to the comments in [protobuf definitions](https://github.com/cosmos/ics23/blob/master/proto/cosmos/ics23/v1/proofs.proto).

Reply:
yes but do you think 
``` 
spec.max_depth < spec.min_depth
            || spec.min_depth < 0
            || spec.max_depth < 0
```
is correct or it should be as I mention?

Reply:
This is not right, as you correctly specified. The error case should be something like this,

```rs
0 < spec.min_depth && 0 < spec.max_depth && spec.max_depth < spec.min_depth
```

So, if both of `min_depth` and `max_depth` are greater than zero, `min_depth <= max_depth`.